src/main.cpp

```diff
@@ -4769,17 +4779,19 @@ static bool AlreadyHave(const CInv& inv) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
     {
     case MSG_TX:
         {
-            assert(recentRejects);
+            // remove assertions from P2P code, but this should hold: assert(recentRejects);
```

Comment:
I'm handling the check below.  As a previously stated, recentrejects is a nice-to-have but I do not think the node should go down if there is an issue with it.